openapi_test.go

```diff
@@ -270,16 +268,14 @@ func TestServer_OutputOpenApiSpec(t *testing.T) {
 	})
 	t.Run("do not print file", func(t *testing.T) {
 		s := NewServer(
-			WithEngineOptions(
+			WithEngine(NewEngine(
```

Comment:
What about a helper like this

```suggestion
			WithNewEngine(
```

Do you really need a new engine BTW?

Reply:
Yes that’s better

Reply:
And what do you mean do I need a “NewEngine”. The whole purpose of this PR is to demonstrate the difference between WithEngineOptions and WithEngine. Do you have another pattern in mind that you would suggest?

Reply:
I guess looking at it. WithEngine is more redundant than the WithEngineOptions. 

I do like the splitting of the config that pertinent to the Engine and pertinent to the net/http server. I think imma update the other to split those out.